JetMETCorrections/Type1MET/interface/PFJetMETcorrInputProducerT.h

```diff
@@ -82,8 +83,10 @@ class PFJetMETcorrInputProducerT : public edm::EDProducer
       offsetCorrLabel_ = cfg.getParameter<edm::InputTag>("offsetCorrLabel");
       offsetCorrToken_ = consumes<reco::JetCorrector>(offsetCorrLabel_);
     }
-    jetCorrLabel_ = cfg.getParameter<edm::InputTag>("jetCorrLabel");
+    jetCorrLabel_ = cfg.getParameter<edm::InputTag>("jetCorrLabel"); //for MC
+    jetCorrLabelRes_ = cfg.getParameter<edm::InputTag>("jetCorrLabelRes"); //for data
     jetCorrToken_ = consumes<reco::JetCorrector>(jetCorrLabel_);
+    jetCorrResToken_ = consumes<reco::JetCorrector>(jetCorrLabelRes_);
```

Comment:
Both ```jetCorrToken_``` and ```jetCorrResToken_``` should use ```mayConsume``` instead of ```consume``` because only one of those two will be called per event.